src/main/java/Parser.java

```diff
@@ -0,0 +1,166 @@
+            break;
+        }
+        return t;
+    }
+
+    /**
+     * Returns a array of LocalDateTime object
+     * @param data String that contains the date
+     * @param errorMessage  error message to display when error is occur during prasing to LocalDateTIme
+     * @return LocalDateTime[].
+     * @throws DukeDeadlineException
+     */
+
+    public static LocalDateTime[] parseDates(String data, String errorMessage) throws DukeDeadlineException {
+        int firstSpace = data.indexOf(" ");
+        String date = data.substring(0,firstSpace);
+        data = data.substring(firstSpace + 1);
+        String[] timePeriod = data.split("-");
+        if (timePeriod.length != 2) {
+            throw new DukeDeadlineException(errorMessage);
+        }
+        LocalDateTime[] deadline = new LocalDateTime[2];
+        for (int i = 0; i < timePeriod.length; i++) {
+            deadline[i] = parseDate(date + " " + timePeriod[i], errorMessage);
+        }
+        return deadline;
+    }
+
+    /**
+     * Returns LocalDateTime object based on string given
+     * @param date String that contains the date in the format of yyyy-M-d hhmm
+     * @param errorMessage error message to display when error is occur during prasing to LocalDateTIme
+     * @return LocalDateTime
+     * @throws DukeDeadlineException
+     */
+    public static LocalDateTime parseDate(String date, String errorMessage) throws DukeDeadlineException{
+        try {
+            return LocalDateTime.parse(date, DateTimeFormatter.ofPattern("yyyy-M-d Hmm"));
+        } catch (DateTimeParseException e) {
+            throw new DukeDeadlineException(errorMessage);
+        }
+    }
+
+    /**
+     * Returns a Task Object based on the csvData provided
+     * @param csvData (a row from the saved file)
+     * @param ui (UI object for pririnting
+     * @return
+     */
+
+    public static Task parseForText(String csvData) throws DukeDeadlineException {
+        String errorMessage = "The deadline for this task is corrupted (Required : yyyy-M-d hhmm";
+        String[] taskArr = csvData.split(",");
+        Task task = null;
+        try {
+            switch (taskArr[0]) {
+            case "T":
+                task = new Todo(taskArr[2]);
+                break;
+            case "E":
+                String[] deadlineArr = {taskArr[3], taskArr[4]};
+                LocalDateTime[] deadlines = new LocalDateTime[2];
+                for (int i = 0; i < deadlineArr.length; i++) {
+                    deadlines[i] = Parser.parseDate(deadlineArr[i], errorMessage);
+                }
+                task = new Event(taskArr[2], deadlines[0], deadlines[1]);
+                break;
+            case "D":
+                LocalDateTime deadline = Parser.parseDate(taskArr[3], errorMessage);
+                task = new Deadline(taskArr[2], deadline);
+            }
+            if (taskArr[1] == "1") {
+                task.markAsDone();
+            }
```

Comment:
Could consider ending the last case statement with a 'break'.

From the CS2103 Java Coding Standard:
The explicit //Fallthrough comment should be included whenever there is a case statement without a break statement.
Rationale: Leaving out the break is a common error, and it must be made clear that it is intentional when it is not there.